### ISSUE TITLE
Add ignoreFunctions: [] to value-keyword-case

### ISSUE THREAD

**Issue by FollowingFawn** [stylelint/stylelint#4648]
> Clearly describe the bug

I am using stylelint and I have a several different custom color variables used in several properties such as border, background, color [using a non-standard `t` function].

> Which rule, if any, is the bug related to?

value-keyword-case

> What code is needed to reproduce the bug?

e.g.

```css
a {
  border: 1px solid t(lightGray);
  background-color: t(lightBlue);
  color: var(--lightBlue);
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
       "value-keyword-case": ["lower", {"ignoreKeywords": ["/^t(*)/"]}],
  }
}
```

> Which version of stylelint are you using?

13.2.1

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

    "force-app/main/default/**/*.css": [
      "stylelint --fix"
    ]

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

I want to either ignore any custom variable inside t() or var(--) or allow camelCasing. 

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
Expected "digitalBlue" to be "digitalblue"   value-keyword-case    
```

Is there a way to either ignore camelCasing or ignore anything within t() or var(--)?, e.g.

```
"value-keyword-case": ["lower", {"ignoreKeywords": ["/^t(*)/"]}]
```

**Comment by jeddy3** [comment]
@FollowingFawn Thanks for the report and for using the template.

> Is there a way to either ignore camelCasing or ignore anything within t() or var(--)?

Keywords within `var` functions are already ignored by the rule.

Given:

```css
a {
  color: var(--camelCase);
  color: t(--camelCase);
}
```

We get:

```
3:12 error Expected "--camelCase" to be "--camelcase" (value-keyword-case)
```

As only the second declaration is reported as a violation. See [this demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICGACYAdAdilkA2EATvCgG5KEAUAtNWEgLYx4DCSAzjAJQDcWO+IiQAuNOo2ZtOvLAF8sIADQh2AT0zCkADwTh27JeAiYAZgEsA5rozYU6EIQCueGO3skbOHPfJ5HMagBrGFUAdyIocU53OxACUJhCe34UeUxZEFkgA).

Keywords within `t` functions aren't ignored because `t` is a non-standard function. I recommend you only use `var` for variables. However, if this is not possible for some reason, we can add an `ignoreFunctions` optional secondary option to the rule:

```json
{
  "rules": {
       "value-keyword-case": ["lower", {"ignoreFunctions": ["t"]}],
  }
}
```

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.

**Comment by malsf21** [comment]
Just submitted a PR for this feature! #4733
